apps/site/lib/site_web/controllers/search_controller.ex

```diff
@@ -72,27 +73,29 @@ defmodule SiteWeb.SearchController do
   end
 
   defp do_query(response, conn) do
-    _ = log_error(response)
+    log_error(response)
+
     json(conn, %{error: "bad_response"})
   end
 
-  @spec log_error({:ok, HTTPoison.Response.t()} | {:error, HTTPoison.Error.t() | atom}) ::
-          :ok | {:error, any}
   def log_error({:ok, %HTTPoison.Response{} = response}) do
-    do_log_error(response)
-  end
-
-  def log_error({:error, %HTTPoison.Error{} = response}) do
-    do_log_error(response)
+    response
+    |> Map.from_struct()
+    |> Map.take([:body, :headers, :status_code])
+    |> Keyword.new(fn {k, v} -> {k, v} end)
```

Comment:
For the `%HTTPoison.Response{}` `body` it appears to be JSON-encoded, example `"{\"message\":\"Invalid value for \\\"query\\\" parameter, expected string shorter than 512 bytes\",\"status\":400}"`...maybe we should just decode and log this instead?

Reply:
I'm not entirely sure how logging a map would work. But, I think it will just flatten out it like this.